designs/Verify-Verbosity.md

```diff
@@ -0,0 +1,302 @@
+  Valid from: 12/22/2017 4:00:00 PM to 3/22/2029 4:59:59 PM
+trace:       Subject Name: CN=Symantec SHA256 TimeStamping CA, OU=Symantec Trust Network, O=Symantec Corporation, C=US
+trace:       SHA1 hash: 6FC9EDB5E00AB64151C1CDFCAC74AD2C7B7E3BE4
+trace:       SHA256 hash: F3516DDCC8AFC808788BD8B0E840BDA2B5E23C6244252CA3000BB6C87170402A
+trace:       Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:       Valid from: 1/11/2016 4:00:00 PM to 1/11/2031 3:59:59 PM
+trace: 
+trace:             Subject Name: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:             SHA1 hash: 3679CA35668772304D30A5FB873B0FA77BB70D54
+trace:             SHA256 hash: 2399561127A57125DE8CEFEA610DDF2FA078B5C8067F4E828290BFB860E84B3C
+trace:             Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:             Valid from: 4/1/2008 5:00:00 PM to 12/1/2037 3:59:59 PM
+
+Successfully verified package 'NuGet.Common.5.9.0-preview.2'.
+```
+
+</details>
+
+#### `--verbosity diagnostic`
+
+* Same output as `--verbosity detailed`
+
+### Example output in failure scenarios
+
+* Errors and Warnings are displayed to the console irrespective of `verbosity` level.
+
+#### `Verifying tampered package`
+
+<details>
+<summary>output</summary>
+
+```
+dotnet nuget verify tampered.12.0.1.nupkg -v -detailed
+error: NU3008: The package integrity check failed.
+error: Package signature validation failed.
+```
+
+</details>
+
+#### `Verifying author signed package with no timestamp and untrusted certificate root`
+
+<details>
+<summary>output</summary>
+
+```
+dotnet nuget verify "package.nupkg" -v n
+Verifying packageA.1.0.0
+C:\Users\kapenaga\Downloads\package.nupkg
+Signature Hash Algorithm: SHA256
+
+Signature type: Author
+Verifying the author primary signature with certificate:
+  Subject Name: CN=test
+  SHA1 hash: B0A2B3B1695AB8361B1D2B14A9F5D64136E26380
+  SHA256 hash: 89A2B6EB529E0AEBF0D11C8A18A846C7B8D1290791B6BF494BAFEC299F2EAAB2
+  Issued by: CN=test
+  Valid from: 1/29/2021 12:28:11 PM to 1/29/2021 1:28:11 PM
+error: NU3018: The author primary signature found a chain building
+issue: UntrustedRoot: A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider.
+error: NU3037: The author primary signature validity period has expired.
+warn : NU3027: The signature should be timestamped to enable long-term signature validity after the certificate has expired.
+Finished with 2 errors and 1 warnings.
+
+error: Package signature validation failed.
+```
+
+</details>
```

Comment:
Can we also include the Debian case before and after the incident:

* Valid author signing certificate + untrusted timestamping certificate (you should see the `NU3027` warning in this case)
* Expired author signing certificate + untrusted timestamping certificate

Reply:
added couple of failure cases related to debian issue to the spec.

Reply:
@loic-sharma - I have updated the spec with Debian case before and after the incident. Please note that, `dotnet nuget verify` or `nuget.exe verify` command operates under [default policy](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignedPackageVerifierSettings.cs#L231-L246) and hence `NU3028: The author primary signature's timestamping certificate is not trusted by the trust provider` is considered as error.

Reply:
Although "the Debian case" refers to an [incident from earlier this year](https://github.com/NuGet/Home/issues/10491) where Symantec CA distrust was first reported on Debian family distributions, we should not make the good name of Debian synonymous with the incident.  It would be better to label this the "Symantec CA distrust case."